File: DSA-3/Recursion/tower-of-hanoi.cpp
void toh(int N, int from, int to, int aux, long long &moves) {
    // Your code here
    if(N == 1){
      moves++;
      cout<< "move disk 1 from rod "<<from<<" to rod "<<to<<endl;
      return;
    }
    
   toh(N-1, from, aux, to, moves);
   moves++;
   cout<< "move disk "<< N <<" from rod "<< from <<" to rod "<< to <<endl;
   toh(N-1, aux, to, from, moves);
}

File: Company Wise Practice/Goldman Sachs/tower-of-hanoi.java
/*package whatever //do not write package name here */

import java.util.*;
import java.lang.*;
import java.io.*;

public class GFG {
    
    public static Integer count = 0;
    
    public static void towerOfHanoi(Integer n,Character from, Character to, Character aux){
        
        if(n <= 0) return;
        
        towerOfHanoi(n-1,from,aux,to);
        System.out.println("move disk "+ n +" from rod "+ from +" to rod "+ aux);
        count++;
        towerOfHanoi(n-1,to,from,aux);
    }
	public static void main (String[] args) {
		//code
		Scanner sc = new Scanner(System.in);
		Integer t = sc.nextInt();
		
		while(t-- > 0){
		    
		    Integer n = sc.nextInt();
		    count = 0;
		    towerOfHanoi(n,'1','2','3');
		    System.out.println(count);
		}
		
		sc.close();
		
	}
}